R/loo_predictive_error.R

```diff
@@ -0,0 +1,202 @@
+#' Estimate leave-one-out predictive errors.
+#'
+#' The `loo_predictive_error()` function computes estimates of leave-one-out
+#' predictive errors given a set of predictions and observations. Curreantly
+#' supported error metrics are mean absolute error, mean squared error and root
+#' mean squared error for continuous predictions and accuracy and balanced
+#' accuracy for binary classification. Predictions are passed on to the
+#' `E_loo()` function, so this function assumes that the PSIS approximation is
+#' working well.
+#'
+#' @param x A numeric matrix of predictions.
+#' @param y A numeric vector of observations. Length should be equal to the
+#'     number of rows in `x`.
+#' @param ll A matrix of pointwise log-likelihoods. Should be of same dimension
+#'     as `x`.
+#' @param pred_error The type of predictive error to be used. Currently
+#'     supported options are `"mae"`, `"rmse"` and `"mse"` for regression and
+#'     for binary classification `"acc"` and `"balanced_acc"`.
+#'     \describe{
+#'       \item{`"mae"`}{
+#'          Mean absolute error.
+#'       }
+#'       \item{`"mse"`}{
+#'          Mean squared error.
+#'       }
+#'       \item{`"rmse"`}{
+#'          Root mean squared error, given by as the square root of `MSE`.
+#'       }
+#'       \item{`"acc"`}{
+#'          The proportion of predictions indicating the correct outcome.
+#'       }
+#'       \item{`"balanced_acc"`}{
+#'          Balanced accuracy is given by the average of true positive and true
+#'          negative rates.
+#'       }
+#'     }
+#' @param r_eff A Vector of relative effective sample size estimates containing
+#'     one element per observation. See [psis()] for more details.
+#' @param ... Additional arguments passed on to [E_loo()]
+#'
+#' @return A list with the following components:
+#' \describe{
+#'   \item{`estimate`}{
+#'   Estimate of the given predictive error measure.
+#'   }
+#'  \item{`se`}{
+#'   Standard error of the estimate.
+#'   }
+#'  }
+#' @export
+#'
+#' @examples
+#' \donttest{
+#' if (requireNamespace("rstanarm", quietly = TRUE)) {
+#' # Use rstanarm package to quickly fit a model and get both a log-likelihood
+#' # matrix and draws from the posterior predictive distribution
+#' library("rstanarm")
+#'
+#' # data from help("lm")
+#' ctl <- c(4.17,5.58,5.18,6.11,4.50,4.61,5.17,4.53,5.33,5.14)
+#' trt <- c(4.81,4.17,4.41,3.59,5.87,3.83,6.03,4.89,4.32,4.69)
+#' d <- data.frame(
+#'   weight = c(ctl, trt),
+#'   group = gl(2, 10, 20, labels = c("Ctl","Trt"))
+#' )
+#' fit <- stan_glm(weight ~ group, data = d, refresh = 0)
+#' ll <- log_lik(fit)
+#' r_eff <- relative_eff(exp(-ll), chain_id = rep(1:4, each = 1000))
```

Comment:
I think the doc needs to be regenerated one more time so that the .Rd file includes this change to `-ll` instead of `log_ratios`. That's why R cmd check is failing but the unit tests are passing.